bigquery_storage/noxfile.py

```diff
@@ -13,113 +15,136 @@
 # limitations under the License.
 
 from __future__ import absolute_import
-
 import os
-import shutil
 
 import nox
 
 
-LOCAL_DEPS = (
-    os.path.join('..', 'api_core'),
-)
+LOCAL_DEPS = (os.path.join("..", "api_core"), os.path.join("..", "core"))
 
+@nox.session(python="3.7")
+def lint(session):
+    """Run linters.
+
+    Returns a failure if the linters find linting errors or sufficiently
+    serious code quality issues.
+    """
+    session.install("flake8", "black", *LOCAL_DEPS)
+    session.run(
+        "black",
+        "--check",
+        "google",
+        "tests",
+        "docs",
+    )
+    session.run("flake8", "google", "tests")
 
-def default(session):
-    """Default unit test session.
 
-    This is intended to be run **without** an interpreter set, so
-    that the current ``python`` (on the ``PATH``) or the version of
-    Python corresponding to the ``nox`` binary on the ``PATH`` can
-    run the tests.
+@nox.session(python="3.6")
+def blacken(session):
+    """Run black.
+
+    Format code to uniform standard.
     """
+    session.install("black")
+    session.run(
+        "black",
+        "google",
+        "tests",
+        "docs",
+    )
+
+
+@nox.session(python="3.7")
+def lint_setup_py(session):
+    """Verify that setup.py is valid (including RST check)."""
+    session.install("docutils", "pygments")
+    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")
+
+
+def default(session):
     # Install all test dependencies, then install this package in-place.
-    session.install('mock', 'pytest', 'pytest-cov')
+    session.install("mock", "pytest", "pytest-cov")
     for local_dep in LOCAL_DEPS:
-        session.install('-e', local_dep)
-    session.install('-e', '.[pandas,fastavro]')
+        session.install("-e", local_dep)
+    session.install("-e", ".")
+    session.install("-e", "pandas")
+    session.install("-e", "fastavro")
```

Comment:
This bit is incorrect:  we should be installing the *extras* of this package.  Note that https://github.com/pypa/pip/issues/4957 prevents getting the extras installed if the host package is already installed.